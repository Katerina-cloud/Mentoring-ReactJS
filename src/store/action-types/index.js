export const MOVIES_LOAD = 'MOVIES_LOAD';
export const MOVIES_SUCCESS = 'MOVIES_SUCCESS';
export const MOVIES_FAIL = 'MOVIES_FAIL';
export const SET_EDIT_MOVIE = 'SET_EDIT_MOVIE';
export const CLEAR_EDIT_MOVIE = 'CLEAR_EDIT_MOVIE';
export const SET_DELETE_MOVIE = 'SET_DELETE_MOVIE';
export const CLEAR_DELETE_MOVIE = 'CLEAR_DELETE_MOVIE';
export const SET_ADD_MOVIE = 'SET_ADD_MOVIE';
export const CLEAR_ADD_MOVIE = 'CLEAR_ADD_MOVIE';
export const SET_FILTER_BY_GENRE = 'SET_FILTER_BY_GENRE';
export const CLEAR_FILTER_BY_GENRE = 'CLEAR_FILTER_BY_GENRE';
export const SET_CARD = 'SET_CARD';

export const DELETE_MOVIE = 'DELETE_MOVIE';
export const DELETE_MOVIE_FAIL = 'DELETE_MOVIE_FAIL';
export const DELETE_MOVIE_SUCCESS = 'DELETE_MOVIE_SUCCESS';
export const EDIT_MOVIE_SUCCESS = 'EDIT_MOVIE_SUCCESS';
export const EDIT_MOVIE = 'EDIT_MOVIE';
export const EDIT_MOVIE_FAIL = 'EDIT_MOVIE_FAIL';
export const ADD_MOVIE_SUCCESS = 'ADD_MOVIE_SUCCESS';
export const ADD_MOVIE = 'ADD_MOVIE';
export const ADD_MOVIE_FAIL = 'ADD_MOVIE_FAIL';
