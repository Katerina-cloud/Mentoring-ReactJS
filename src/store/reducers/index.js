import { combineReducers } from 'redux';
import { moviesReducer } from './movies';

export const appReducer = combineReducers({
  moviesData: moviesReducer,
});
