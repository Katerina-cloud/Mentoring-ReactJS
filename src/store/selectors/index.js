export { moviesSelector } from './movies';
export { filterGenreSelector } from './filterGenre';
export { editMovieSelector } from './editMovie';
export { deleteMovieSelector } from './deleteMovie';
export { addMovieSelector } from './addMovie';
export { sortParameterSelector } from './sortParameter';
