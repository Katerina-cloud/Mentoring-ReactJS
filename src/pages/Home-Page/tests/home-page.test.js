import React from 'react';
import { render, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { HomePage } from '../';
import { Provider } from 'react-redux';
import { store } from '../../../store/index';

describe('HomePage', () => {
  it('snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <HomePage />
      </Provider>,
      { wrapper: MemoryRouter },
    );

    expect(tree).toMatchSnapshot();
  });
});
