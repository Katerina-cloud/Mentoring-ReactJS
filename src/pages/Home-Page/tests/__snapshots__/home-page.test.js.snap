// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="header__add-movie"
        >
          <div
            class="logo"
          >
            <span
              class="logo--bold"
            >
              netflix
            </span>
            <span>
              roulette
            </span>
          </div>
          <div
            class="header__button"
          >
            <button
              class="button button--text-red button--gray button--border-undefined"
            >
              +Add movie
            </button>
          </div>
        </div>
        <div
          class="header__search-bar"
        >
          <div
            class="search-bar"
          >
            <h1
              class="search-bar__title"
            >
              Find your movie
            </h1>
            <div
              class="search-bar__search-container"
            >
              <div
                class="search-bar__form-input"
              >
                <input
                  class="form-input__text"
                  placeholder="What do you want to watch?"
                  value=""
                />
              </div>
              <div
                class="search-bar__search-button"
              >
                <button
                  class="button button--text-undefined button--red button--border-undefined"
                >
                  Search
                </button>
              </div>
            </div>
          </div>
        </div>
      </header>
      <main
        class="home-page__main"
      >
        <div
          class="home-page__filter"
        >
          <div
            class="filter-bar__wrapper"
          >
            <div
              class="filter-bar"
            >
              <div
                class="filter-bar__buttons"
              >
                <div
                  class="filter-bar__button"
                >
                  <button
                    class="filter-button"
                  >
                    All
                  </button>
                </div>
                <div
                  class="filter-bar__button"
                >
                  <button
                    class="filter-button"
                  >
                    Documentary
                  </button>
                </div>
                <div
                  class="filter-bar__button"
                >
                  <button
                    class="filter-button"
                  >
                    Comedy
                  </button>
                </div>
                <div
                  class="filter-bar__button"
                >
                  <button
                    class="filter-button"
                  >
                    Horror
                  </button>
                </div>
                <div
                  class="filter-bar__button"
                >
                  <button
                    class="filter-button"
                  >
                    Crime
                  </button>
                </div>
              </div>
              <div
                class="dropdown"
              >
                <label
                  class="dropdown__label"
                  for="films-sort"
                >
                  Sort by
                </label>
                <select
                  class="dropdown__select"
                >
                  <option
                    class="dropdown__option"
                    value="release_date"
                  >
                    release date
                  </option>
                  <option
                    class="dropdown__option"
                    value="vote_average"
                  >
                    rating
                  </option>
                </select>
              </div>
            </div>
            <div
              class="filter-bar__border"
            />
            <div
              class="filter-bar__results"
            >
              <span
                class="filter-bar__results--bold"
              >
                0
              </span>
               movies found
            </div>
          </div>
        </div>
        <div
          class="home-page__films"
        />
      </main>
      <footer
        class="footer"
      >
        <div
          class="logo"
        >
          <span
            class="logo--bold"
          >
            netflix
          </span>
          <span>
            roulette
          </span>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="header__add-movie"
      >
        <div
          class="logo"
        >
          <span
            class="logo--bold"
          >
            netflix
          </span>
          <span>
            roulette
          </span>
        </div>
        <div
          class="header__button"
        >
          <button
            class="button button--text-red button--gray button--border-undefined"
          >
            +Add movie
          </button>
        </div>
      </div>
      <div
        class="header__search-bar"
      >
        <div
          class="search-bar"
        >
          <h1
            class="search-bar__title"
          >
            Find your movie
          </h1>
          <div
            class="search-bar__search-container"
          >
            <div
              class="search-bar__form-input"
            >
              <input
                class="form-input__text"
                placeholder="What do you want to watch?"
                value=""
              />
            </div>
            <div
              class="search-bar__search-button"
            >
              <button
                class="button button--text-undefined button--red button--border-undefined"
              >
                Search
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      class="home-page__main"
    >
      <div
        class="home-page__filter"
      >
        <div
          class="filter-bar__wrapper"
        >
          <div
            class="filter-bar"
          >
            <div
              class="filter-bar__buttons"
            >
              <div
                class="filter-bar__button"
              >
                <button
                  class="filter-button"
                >
                  All
                </button>
              </div>
              <div
                class="filter-bar__button"
              >
                <button
                  class="filter-button"
                >
                  Documentary
                </button>
              </div>
              <div
                class="filter-bar__button"
              >
                <button
                  class="filter-button"
                >
                  Comedy
                </button>
              </div>
              <div
                class="filter-bar__button"
              >
                <button
                  class="filter-button"
                >
                  Horror
                </button>
              </div>
              <div
                class="filter-bar__button"
              >
                <button
                  class="filter-button"
                >
                  Crime
                </button>
              </div>
            </div>
            <div
              class="dropdown"
            >
              <label
                class="dropdown__label"
                for="films-sort"
              >
                Sort by
              </label>
              <select
                class="dropdown__select"
              >
                <option
                  class="dropdown__option"
                  value="release_date"
                >
                  release date
                </option>
                <option
                  class="dropdown__option"
                  value="vote_average"
                >
                  rating
                </option>
              </select>
            </div>
          </div>
          <div
            class="filter-bar__border"
          />
          <div
            class="filter-bar__results"
          >
            <span
              class="filter-bar__results--bold"
            >
              0
            </span>
             movies found
          </div>
        </div>
      </div>
      <div
        class="home-page__films"
      />
    </main>
    <footer
      class="footer"
    >
      <div
        class="logo"
      >
        <span
          class="logo--bold"
        >
          netflix
        </span>
        <span>
          roulette
        </span>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
